Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -35,12 +35,13 @@ class Navigator {
      - parameter completionHandler: A block object to be executed when history dumping ends.
      */
     func writeHistory(completionHandler: @escaping (URL?) -> Void) {
-        historyRecorder?.dumpHistory { (path) in
+        historyRecorder?.stopRecording { [weak self] (path) in
             if let path = path {
                 completionHandler(URL(fileURLWithPath: path))
             } else {
                 completionHandler(nil)
             }
+            self?.historyRecorder?.startRecording()
```

Comment:
According to the `HistoryRecorderHandle.stopRecording(result:)` documentation, we have to explicitly restart history recording after it’s done writing out history, a change from before when `dumpHistory(result:)` would keep history recording going.

Reply:
Unfortunately, this change seems to be causing [various test failures](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/4377/workflows/14112c3c-9914-414b-a086-b3e988d70685/jobs/38325):

MapboxCoreNavigationTests.PassiveLocationManagerTests
  testHistoryRecording, XCTAssertNotNil failed
  /Users/distiller/project/Tests/MapboxCoreNavigationTests/PassiveLocationManagerTests.swift:124
  ```
        PassiveLocationManager.writeHistory { url in
            XCTAssertNotNil(url)
            historyExpectation.fulfill()
  ```

Reply:
#3157 moves this `startRecording()` call outside the callback to happen synchronously with `stopRecording(result:)`, as recommended by @DmitryAk.